companion/gateway/services/groq_client.py

```
@@ -0,0 +1,450 @@
+                    "max_tokens": max_tokens,
+                    "temperature": temperature
+                }
+            )
+            
+            if response.status_code == 200:
+                return response.json()
+            else:
+                raise ServiceUnavailableError(
+                    service_name="Groq",
+                    message=f"API request failed with status {response.status_code}: {response.text}"
+                )
```

Ruff (0.13.2)

Abstract `raise` to an inner function

(TRY301)